hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -355,6 +356,22 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 		return ctrl.Result{}, fmt.Errorf("failed to look up release image metadata: %w", err)
 	}
 
+	// Validate the NodePool and management cluster CPU architectures match if the HostedCluster does not support multi-arch NodePools
+	supportsMultiArchNodePoolCondition := meta.FindStatusCondition(hcluster.Status.Conditions, hyperv1.SupportsMultiArchNodePools)
+	if supportsMultiArchNodePoolCondition != nil && supportsMultiArchNodePoolCondition.Status == metav1.ConditionFalse {
+		// The HostedCluster does not support managing different CPU architecture NodePools. Validate the CPU
+		// architecture of the management cluster and the NodePool match.
+		mgmtClusterCPUArch, err := supportutil.GetMgmtClusterCPUArch(ctx)
```

Comment:
there's a client already built for this reconciler this should be using instead of let this func recreate one every time